services/wireguard/endpoint/create_linux.go

```diff
@@ -20,15 +20,27 @@
 package endpoint
 
 import (
+	log "github.com/cihub/seelog"
 	"github.com/mysteriumnetwork/node/core/ip"
 	wg "github.com/mysteriumnetwork/node/services/wireguard"
 	"github.com/mysteriumnetwork/node/services/wireguard/endpoint/kernelspace"
+	"github.com/mysteriumnetwork/node/services/wireguard/endpoint/userspace"
 	"github.com/mysteriumnetwork/node/services/wireguard/resources"
+	"github.com/mysteriumnetwork/node/utils"
 )
 
 // NewConnectionEndpoint creates new wireguard connection endpoint.
 func NewConnectionEndpoint(ipResolver ip.Resolver, resourceAllocator *resources.Allocator) (wg.ConnectionEndpoint, error) {
-	wgClient, err := kernelspace.NewWireguardClient()
+	var wgClient wgClient
+	var err error
+
+	if isKernelSpaceSupported() {
```

Comment:
maybe extract this to a separate function, something like GetWGClient?

Reply:
Extracted it.